src/drivers/intel/ssp/ssp.c

```diff
@@ -784,10 +811,12 @@ static void ssp_start(struct dai *dai, int direction)
 
```

Comment:
since bclk_delay is added here, it seems to me that TSRE/RSRE bit is irrelevant to the clock itself.

Reply:
> since bclk_delay is added here, it seems to me that TSRE/RSRE bit is irrelevant to the clock itself.

I will check the spec, thanks.

Reply:
@bardliao have you checked if TSRE/RSRE set is mandatory/necessary for BCLK launching? or the setting of SSCR0_SSE? which is the crucial one?

Reply:
> @bardliao have you checked if TSRE/RSRE set is mandatory/necessary for BCLK launching? or the setting of SSCR0_SSE? which is the crucial one?

**SSCR0_SSE** is the crucial one.

Reply:
When I set SSE bit, I see both bclk and fsync on my scope. @bardliao is it the same on your side?

Reply:
We are told to enable TSRE and RSRE long before SSE, so that the fifos are filled.
That said, this is not really working because we don't seem to have a pipeline pre-roll so we still have a risk of FIFOs not being full before enabling the SSE bit. That will have to be another PR...

Reply:
> When I set SSE bit, I see both bclk and fsync on my scope. @bardliao is it the same on your side?

No, only bclk on my side. I tested on Up Xtreme (TGL) board. It looks like fsync starts when DMA starts.

Reply:
> We are told to enable TSRE and RSRE long before SSE, so that the fifos are filled.
> That said, this is not really working because we don't seem to have a pipeline pre-roll so we still have a risk of FIFOs not being full before enabling the SSE bit. That will have to be another PR...

But look at the existing [ssp_start](https://github.com/thesofproject/sof/blob/main/src/drivers/intel/ssp/ssp.c#L746) we enable TSRE and RSRE after SSE.

Reply:
I know @bardliao and hardware folks tell us it's wrong.
The trigger sequence needs to be revisited in a separate PR.

Reply:
Sorry I am wrong, the SSE bit itself only enables bclk.